lib/ansible/modules/system/parted.py

```diff
@@ -0,0 +1,654 @@
+    if rc != 0 and 'unrecognised disk label' in out.lower():
+        return True
+
+    return False
+
+
+def parted_version():
+    """
+    Returns the major and minor version of parted installed on the system.
+    """
+    global module
+
+    rc, out, err = module.run_command("parted --version")
+    if rc != 0:
+        module.fail_json(
+            msg="Failed to get parted version.", rc=rc, out=out, err=err
+        )
+
+    lines = out.split('\n')
+    if len(lines) == 0:
+        module.fail_json(msg="Failed to get parted version.", rc=0, out=out)
+
+    matches = re.search(r'^parted.+(\d+)\.(\d+)$', lines[0])
+    if len(matches.groups()) != 2:
+        module.fail_json(msg="Failed to get parted version.", rc=0, out=out)
+
+    major = int(matches.group(1))
+    minor = int(matches.group(2))
+
+    return major, minor
+
+
+def parted(script, device, align):
+    """
+    Runs a parted script.
+    """
+    global module
+
+    if script and not module.check_mode:
+        command = "parted -s -m -a {0} {1} -- {2}".format(align, device, script)
+        rc, out, err = module.run_command(command)
+
+        if rc != 0:
+            module.fail_json(
+                msg="Error while running parted script: '{0}'".format(
+                    command.strip()
+                ),
+                rc=rc, out=out, err=err
+            )
+
+
+def part_exists(partitions, attribute, number):
+    """
+    Looks if a partition that has a specific value for a specific attribute
+    actually exists.
+    """
+    return any(
+        part[attribute] and
+        part[attribute] == number for part in partitions
+    )
+
+
+def parse_unit(size_str, unit, ceil=True):
+    """
+    Converts '123.1unit' string into ints. If ceil is True it will be rounded up
+    (124) and and down (123) if ceil is False.
+    """
+    matches = re.search(r'^([\d.]+)(\w+)?$', size_str)
+    if matches is None:
+        module.fail_json(msg="Error interpreting the size of the disk given by parted.")
+
+    size = locale.atof(matches.group(1))
+    if len(matches.groups()) == 2:
+        unit = matches.group(2).lower()
+
+    size = math.floor(size)
+    if ceil:
+        math.ceil(size)
+
+    return size, unit
+
+
+def main():
+    global module
+    changed = False
+    output_script = ""
+    script = ""
+
+    module = AnsibleModule(
+        argument_spec={
+            'device': {'required': True, 'type': 'str'},
+            'align': {
+                'default': 'optimal',
+                'choices': ['none', 'cylinder', 'minimal', 'optimal'],
+                'required': False,
+                'type': 'str'
+            },
+            'number': {'default': None, 'type': 'int', 'required': False},
+
+            # unit <unit> command
+            'unit': {
+                'default': 'KiB',
+                'choices': [
+                    's', 'B', 'KB', 'KiB', 'MB', 'MiB', 'GB', 'GiB', 'TB',
+                    'TiB', '%', 'cyl', 'chs', 'compact'
```

Comment:
I would simply use units_si and units_iec here.
```python
            'choices': units_si + units_iec + ['', 'compact', 'cyl', 'chs'],
```
In fact, make these global variables anyhow.